validation/src/main/scala/hmda/validation/api/ValidationApi.scala

```diff
@@ -0,0 +1,24 @@
+package hmda.validation.api
+
+import hmda.validation.dsl.{ Failure, Result, Success }
+import hmda.validation.engine.ValidationError
+import scalaz._
+import scalaz.Scalaz._
+
+trait ValidationApi {
+
+  def convertResult[T](input: T, result: Result, ruleName: String): ValidationNel[ValidationError, T] = {
+    result match {
+      case Success() => input.success
+      case Failure(msg) => ValidationError(s"$ruleName  failed: $msg").failure.toValidationNel
+    }
+  }
+
+  def validateAll[E, T](checks: List[ValidationNel[E, T]], t: T): ValidationNel[E, T] = {
```

Comment:
@schbetsy I've renamed the variable as per your suggestion, but `t` in this context could be read as "type". I can see how that can be confused with Transmittal Sheet, so I'm fine with changing it